salt/states/dockerng.py

```diff
@@ -421,7 +421,25 @@ def _image_get(path, default=None):
                     data = dict((k, '') for k in data)
                 image_labels.update(data)
                 data = image_labels
-            if actual_data != data:
+
+            def _filter_swarm_labels(_api_data):
```

Comment:
Here's a bit more background.  The labels for a Docker Swarm standalone container could look something like this:

```json
{
  "com.docker.swarm.affinities": "[\"container!=~/myapp.*/\"]",
  "com.docker.swarm.id": "121b791058b8334c59d64b7ccbc5e73099e9b845cbf873bb69b48e6d32652fe5",
  "com.example.deployment_type": "development"
}
```

The idea is that `com.docker.swarm.id` doesn't need to be compared when seeing if a new container is required.

I honestly can't recall when the "empty" (`[]`) `com.docker.swarm.affinities` value could come up, but when I wrote this several months ago, I did need it.